### PR TITLE
To delete synapse library from tmp directory

### PR DESCRIPTION
**Pull request by Shakila**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/synapse/deployers/LibraryArtifactDeployer.java [hunk 1]
```diff
@@ -149,6 +148,9 @@ public void undeploy(String fileName) throws DeploymentException {
 
 	if (deploymentStore.containsFileName(fileName)) {
 	    File undeployingFile = new File(fileName);
+		if(fileName.contains("/tmp/carbonapps/")){
```
Comment (malakasilva) [inline modules/core/src/main/java/org/apache/synapse/deployers/LibraryArtifactDeployer.java]: I think there are some checks needs to be done before deleting the files. like file exists etc. Don't you need to check for file name *.zip. Have you tested by including some other artifacts in your car. rg: proxy api sequence and class mediator?

Comment (Shakila) [inline modules/core/src/main/java/org/apache/synapse/deployers/LibraryArtifactDeployer.java]: Yes, I included other artifacts and tested with the CApp. It's working fine.

File: modules/core/src/main/java/org/apache/synapse/deployers/LibraryArtifactDeployer.java [hunk 2]
```diff
@@ -59,47 +59,50 @@ public void deploy(DeploymentFileData deploymentFileData) throws DeploymentExcep
...
-	    } else {
-		// new artifact hot-deployment case
-		try {
-		    // add the library to synapse Config
-		    completeDeployment(lib, libArtifactName);
-		} catch (SynapseArtifactDeploymentException sade) {
-		    log.error("Deployment of the Synapse Artifact from file : " + libFilePath
-			    + " : Failed!", sade);
+			if (deploymentStore.isUpdatingArtifact(libFilePath)) {
+
```
Comment (malakasilva) [inline modules/core/src/main/java/org/apache/synapse/deployers/LibraryArtifactDeployer.java]: Unnecessary spacing

File: modules/core/src/main/java/org/apache/synapse/deployers/LibraryArtifactDeployer.java [hunk 3]
```diff
@@ -59,47 +59,50 @@ public void deploy(DeploymentFileData deploymentFileData) throws DeploymentExcep
...
+				// this is an hot-update case
+				String existingArtifactName = deploymentStore
+						.getUpdatingArtifactWithFileName(libFilePath);
+				deploymentStore.removeUpdatingArtifact(libFilePath);
+				undeploySynapseArtifact(existingArtifactName);
+
+				// deploy from beginning
+				// add the library to synapse Config
+				completeDeployment(lib, libArtifactName);
+
```
Comment (malakasilva) [inline modules/core/src/main/java/org/apache/synapse/deployers/LibraryArtifactDeployer.java]: No need to keep a space here

Comment (Shakila) [inline modules/core/src/main/java/org/apache/synapse/deployers/LibraryArtifactDeployer.java]: https://github.com/wso2/wso2-synapse/pull/370

File: modules/core/src/main/java/org/apache/synapse/deployers/LibraryArtifactDeployer.java [hunk 4]
```diff
@@ -149,6 +151,9 @@ public void undeploy(String fileName) throws DeploymentException {
 
 	if (deploymentStore.containsFileName(fileName)) {
 	    File undeployingFile = new File(fileName);
+		if(fileName.contains("/tmp/carbonapps/") && fileName.endsWith(".zip")){
```
Comment (RKathees) [inline modules/core/src/main/java/org/apache/synapse/deployers/LibraryArtifactDeployer.java]: Use File.separator to support with the other operating system.

Comment (Shakila) [inline modules/core/src/main/java/org/apache/synapse/deployers/LibraryArtifactDeployer.java]: https://github.com/wso2/wso2-synapse/pull/370/

File: modules/core/src/main/java/org/apache/synapse/deployers/LibraryArtifactDeployer.java (hunk 2 above)
Comment (Shakila) [inline modules/core/src/main/java/org/apache/synapse/deployers/LibraryArtifactDeployer.java]: https://github.com/wso2/wso2-synapse/pull/370/

**Pull request merged**